src/ert/__main__.py

```diff
@@ -191,6 +192,28 @@ def run_lint_wrapper(args: Namespace, _: ErtPluginManager) -> None:
     lint_file(args.config)
 
 
+class DeprecatedAction(argparse.Action):
```

Comment:
Tried running  `(py3.11) [faar@be-lx138036 ert]$ ert ensemble_smoother --target-case ff1 test-data/poly_example/poly.ert 
/usr/lib64/python3.11/argparse.py:1978: UserWarning: --target-case is deprecated and will be removed in future versions. Use --target-ensemble instead.
  action(self, namespace, argument_values, option_string)` The last line with action(...) is not really needed

Reply:
Fixed by settings stacklevel=1:

```python
        warnings.warn(
            f"{option_string} is deprecated and will be removed in future versions. {alternative_msg}",
            stacklevel=1,
        )
```